detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ConfigProperty.kt

```
@@ -129,6 +123,38 @@ private fun <T : Any> getValueOrDefault(configAware: ConfigAware, propertyName:
     }
 }
 
+private fun ConfigAware.getListOrDefault(propertyName: String, defaultValue: List<*>): List<String> {
+    return if (defaultValue.all { it is String }) {
+        val defaultValueAsListOfStrings = defaultValue as List<String>
+        valueOrDefaultCommaSeparated(propertyName, defaultValueAsListOfStrings)
+    } else {
+        error("Only lists of strings are supported. '$propertyName' is invalid. ")
+    }
+}
+
+private fun ConfigAware.getExplainedValuesOrDefault(
+    propertyName: String,
+    defaultValue: ExplainedValues
+): ExplainedValues {
+    val valuesAsList: List<*> = valueOrNull(propertyName) ?: return defaultValue
+    if (valuesAsList.all { it is String }) {
+        return ExplainedValues(values = valuesAsList.map { ExplainedValue(it.toString()) })
+    }
+    if (valuesAsList.all { it is Map<*, *> }) {
+        return ExplainedValues(
+            valuesAsList
+                .map { it as Map<*, *> }
+                .map { dict ->
+                    ExplainedValue(
+                        value = dict["value"].toString(),
+                        reason = dict["reason"]?.toString()
```

`.toString()`  or `as String`? I think that `as String` is better so we can fail fast.

---

I see your point. I changed it to throw a ConfigurationError in those cases